src/modules/SX126x/SX126x.cpp

```diff
@@ -443,7 +443,91 @@ int16_t SX126x::startTransmit(uint8_t* data, size_t len, uint8_t addr) {
 }
 
 int16_t SX126x::startReceive(uint32_t timeout) {
-  // set DIO mapping
+  int16_t state = startReceiveCommon();
+  if(state != ERR_NONE) {
+    return(state);
+  }
+
+  // set mode to receive
+  state = setRx(timeout);
+
+  return(state);
+}
+
+int16_t SX126x::startReceiveDutyCycle(uint32_t rxPeriod_us, uint32_t sleepPeriod_us)
+{
+  // datasheet claims time to go to sleep is ~500us, same to wake up.
+  // compensate for that 1ms + tcxo delay:
+  uint32_t transitionTime = _tcxoDelay_us + 1000;
+  if (sleepPeriod_us < transitionTime + 16) {
+    return(ERR_INVALID_SLEEP_PERIOD);
```

Comment:
ERR_SLEEP_PERIOD_TOO_SHORT might be a better name for the error (since it can be too short here, or too long and overflow).

Reply:
As below: How about just having the too long have the same error (but error changes depending on variable), for consistency with the rest of the library?